src/intrinsic.rs

```diff
@@ -341,8 +368,9 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
                 let a = this.read_scalar(args[0])?.to_f32()?;
                 let b = this.read_scalar(args[1])?.to_f32()?;
                 let c = this.read_scalar(args[2])?.to_f32()?;
+                let res = (a*b).value + c;
                 this.write_scalar(
-                    Scalar::from_f32(a * b + c),
+                    Scalar::from_f32(res.value),
```

Comment:
@eddyb is this correct for FMA or would we technically have to avoid the rounding of the intermediate result?

Reply:
"FMA" is a primitive and you cannot implement it with `*` and `+`, you want `mul_add_r`.

Reply:
Ah, I hadn't see that method, thanks. I think I fixed this.